src/hotspot/share/utilities/stableValue.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+* Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+* DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+*
+* This code is free software; you can redistribute it and/or modify it
+* under the terms of the GNU General Public License version 2 only, as
+* published by the Free Software Foundation.
+*
+* This code is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+* version 2 for more details (a copy is included in the LICENSE file that
+* accompanied this code).
+*
+* You should have received a copy of the GNU General Public License version
+* 2 along with this work; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+*
+* Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+* or visit www.oracle.com if you need additional information or have any
+* questions.
+*
+*/
+
+#ifndef SHARE_UTILITIES_STABLEVALUE_HPP
+#define SHARE_UTILITIES_STABLEVALUE_HPP
+
+#include "globalDefinitions.hpp"
+
+// A memory area with adequate size and alignment for storage of a T.
+// May be initialized exactly once.
+template<typename T>
+class StableValue {
+  union {
+    T _t;
+  };
+
+  DEBUG_ONLY(bool _initialized);
+
+public:
+  StableValue() {
+    DEBUG_ONLY(_initialized = false);
+  }
+
+  ~StableValue() {
+  }
+
+  T* ptr() {
+    assert(_initialized, "must be initialized before access");
+    return &this->_t;
+  }
+
+  T* operator->() {
+    assert(_initialized, "must be initialized before access");
+    return ptr();
+  }
+
+  template<typename... Ts>
+  void initialize(Ts&... args) {
+    DEBUG_ONLY(_initialized = true);
+    // If T has const and volatile, get rid of them and tack on a pointer *.
+    using NCVP = std::add_pointer_t<std::remove_cv_t<T>>;
```

Comment:
I think the comment is grammatically confusing as written.  I also think the comment doesn't add any
information, and we'd be better off just removing that comment.